src/components/search/search.component.tsx

```diff
@@ -67,221 +68,229 @@ export interface SearchProps extends ValidationProps, MarginProps {
-        target: {
-          ...(name && { name }),
-          ...(id && { id }),
-          value: "",
+    if (searchButton && onClick) {
+      buttonProps = {
+        onClick: () => {
+          onClick({
+            target: {
+              name,
+              id,
+              value: !isControlled ? searchValue : value,
+            },
+          });
         },
-      });
+      };
     }
-  };
 
-  const handleMouseDown = (event: React.MouseEvent<HTMLElement>) => {
-    event.preventDefault();
-  };
+    const handleIconClick = () => {
+      setSearchValue("");
+      if (onChange) {
+        onChange({
+          target: {
+            ...(name && { name }),
+            ...(id && { id }),
+            value: "",
+          },
+        });
+      }
+    };
 
-  const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
-    setIsFocused(false);
+    const handleMouseDown = (event: React.MouseEvent<HTMLElement>) => {
+      event.preventDefault();
+    };
 
-    if (onBlur) {
-      onBlur(event);
-    }
-  };
+    const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+      setIsFocused(false);
 
-  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
-    if (event.key.length === 1) {
-      event.stopPropagation();
-    }
+      if (onBlur) {
+        onBlur(event);
+      }
+    };
 
-    if (onKeyDown) {
-      onKeyDown(event);
-    }
-  };
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (event.key.length === 1) {
+        event.stopPropagation();
+      }
 
-  const assignInput = useCallback(
-    (input: HTMLInputElement | null) => {
-      if (inputRef) {
-        inputRef.current = input;
+      if (onKeyDown) {
+        onKeyDown(event);
       }
-    },
-    [inputRef]
-  );
+    };
+
+    const isSearchTyped =
+      isFocused || (!isControlled ? !!searchValue.length : !!value.length);
 
-  const isSearchTyped =
-    isFocused || (!isControlled ? !!searchValue.length : !!value.length);
+    return (
+      <StyledSearch
+        isFocused={isFocused}
+        searchWidth={searchWidth}
+        maxWidth={maxWidth}
+        searchIsActive={searchIsActive}
+        searchHasValue={!isControlled ? !!searchValue?.length : !!value?.length}
+        showSearchButton={searchButton}
+        variant={variant}
+        mb={0}
+        {...filterStyledSystemMarginProps(rest)}
+        {...tagComponent("search", rest)}
+        id={id}
+        name={name}
+        {...rest}
+      >
+        <Textbox
+          placeholder={placeholder}
+          value={!isControlled ? searchValue : value}
+          inputIcon={iconType}
+          iconTabIndex={iconTabIndex}
+          iconOnClick={handleIconClick}
+          iconOnMouseDown={handleMouseDown}
+          aria-label={ariaLabel}
+          onFocus={handleFocus}
+          onBlur={handleBlur}
+          onChange={handleChange}
+          onKeyDown={handleKeyDown}
+          ref={ref || inputRef}
```

Comment:
Worth taking a look at #5769 if you get a chance as I've had to make some changes to how the MenuItem passes a ref to Search using context. I don't think it will create many issues but we'll likely need to use a callback so the ref is registered correctly to the item. The linked PR is obviously in review so may change so not suggesting any updates at this stage but may be needed if it's merged before this one